src/main/java/org/springframework/data/elasticsearch/core/RequestFactory.java

```diff
@@ -576,6 +576,10 @@ public IndexRequest indexRequest(IndexQuery query, IndexCoordinates index) {
 			indexRequest.setIfPrimaryTerm(query.getPrimaryTerm());
 		}
 
+		if (query.getRouting() != null) {
```

Comment:
the routing must be added as well to the method `indexRequestBuilder` for the transport client.